src/components/axis.ts

```diff
@@ -96,26 +124,50 @@ export class Axis {
   };
 
   render(): void {
-    const xAxis = axisBottom(this.scaleX);
-    const yAxis = axisRight(this.scaleY);
-
-    const [, width] = this.scaleX.range();
-    const [, height] = this.scaleY.range();
-
-    const gx = this.renderGx(this.mainGroup, xAxis, height);
-    const gy = this.renderGy(this.mainGroup, yAxis, width);
-
-    this.renderLabelx(gx, this.showLabels, width, `${this.labelXDesc} (${this.measurement})`);
-    this.renderLabely(gy, this.showLabels, height, `${this.labelYDesc} (${this.measurement})`);
+    this.renderLabelx();
+    this.renderLabely();
   }
 
   onResize(event: OnResizeEvent): void {
     this.mainGroup.attr('height', `${event.height}px`).attr('width', `${event.width}px`);
   }
 
-  onRescale(event: any): void {
-    this.scaleX = event.xScale;
-    this.scaleY = event.yScale;
+  onRescale(event: OnRescaleEvent): void {
+    const { _scaleX, _scaleY, offsetX, offsetY } = this;
+    const { xScale, yScale } = event;
+    const xBounds = xScale.domain();
+    const yBounds = yScale.domain();
+
+    const xRange = xScale.range();
+    const yRange = yScale.range();
+
+    _scaleX.domain([xBounds[0] - offsetX, xBounds[1] - offsetX]).range(xRange);
+    _scaleY.domain([yBounds[0] - offsetY, yBounds[1] - offsetY]).range(yRange);
+
     this.render();
   }
+
+  get offsetX(): number {
+    return this._offsetX;
+  }
+
+  set offsetX(offset: number) {
+    this._offsetX = offset;
+  }
+
+  get offsetY(): number {
+    return this._offsetY;
+  }
+
+  set offsetY(offset: number) {
+    this._offsetY = offset;
+  }
+
+  get scaleX(): ScaleLinear<number, number> {
+    return this._scaleX.copy();
+  }
+
+  get scaleY(): ScaleLinear<number, number> {
```

Comment:
Is this function used/neccesary? If it is, does it need to copy the scale as the code is now?